src/redisearch_rs/encode_decode/src/qint.rs

```diff
@@ -0,0 +1,284 @@
+//! # qint encoding and decoding
+//!
+//! The qint encoding scheme is a variable-length encoding scheme for integers. It uses a leading byte to
+//! define the number of bytes used to represent the following integers. Based on that leading byte, up to four
+//! variable-length integers are encoded. The leading byte encodes each length in a 2-bit field. The first 2 bits 
+//! represent the first integer. The next 2 bits represent the second integer, and so on.
+//!
+//! The module provides the same function as the C API, but in Rust: [qint_encode2], [qint_encode3], [qint_encode4] and
+//! [qint_decode2], [qint_decode3], [qint_decode4] based on the traits [Read], [Seek] and [Write].
+//!
+//! These methods build upon the generic [qint_encode] and [qint_decode] methods.
+
+
+use std::io;
+
+use std::io::Read;
+use std::io::Seek;
+use std::io::SeekFrom;
+use std::io::Write;
+
+/// Encodes an array of integers into a QInt buffer.
+///
+/// # Arguments
+/// * `cursor` - Buffer writer
+/// * `values` - Array of integers to encode (2, 3, or 4 integers)
+///
+/// # Returns
+/// The number of bytes written to the buffer or an io error
+#[inline(always)]
+pub(crate) fn qint_encode<const N: usize, W>(
+    cursor: &mut W,
+    values: [u32; N],
+) -> Result<usize, std::io::Error>
```

Comment:
The output is not a "plain" usize, is it? 
We should be introducing a new `QIntBuffer` type to represent this concept.